workloadDriver/src/main/java/com/vmware/weathervane/workloadDriver/common/statistics/statsCollector/PerWorkloadStatsCollector.java

```diff
@@ -93,22 +92,12 @@ public PerWorkloadStatsCollector(List<StatsIntervalSpec> statsIntervalSpecs, Loa
 	public void submitOperationStats(OperationStats operationStats) {
 		logger.debug("submitOperationStats: " + operationStats);
 
-		/*
-		 * Add the operationStats to the list of current stats for this period.
-		 * The read lock prevents the list from being replaced while this sample
-		 * is added.  The list is replaced when a statsInterval completes.
-		 */
-		curStatsRWLock.readLock().lock();
-		logger.debug("submitOperationStats locked readLock: " + operationStats);
-		try {
-			if (operationStats != null) {
+		if (operationStats != null) {
+			synchronized (curStatsListLock) {
 				curStatsList.add(operationStats);
-			} else {
-				logger.warn("submitOperationStats for workload {}, received null operationStats", workloadName);
 			}
-		} finally {
-			curStatsRWLock.readLock().unlock();
-			logger.debug("submitOperationStats unlocked readLock: " + operationStats);
+		} else {
+			logger.warn("submitOperationStats for workload {}, received null operationStats", workloadName);
```

Comment:
Should this be an error vs a warning? operationStats shouldn't be null

Reply:
And likewise for the other null operationStats warning below (which I can't leave a comment on)

Reply:
I think it's the same either way for now, since the logger setting is warn. leaving it warn for now -- should probably just remove in future since it (in theory) can't happen).
I don't want to touch the one below since I didn't study the code and don't want to break anything.